Ability to set child ClassName for Gridfield

sebastiand opened the pull request:
Developer can add a config option to use a specific child classname in the gridfield, rather than just SiteTree. 
This allows the gridfield to make use of the child class's `$default_sort` - as we know we only have a single class in GridField.
If we want to manage multiple classes, we skip the $child_classname and the GridField will render children of type SiteTree (as before).

This has many advantages:
* `$default_sort` of child class is picked up
* `GridFieldSortableHeader` will work with all variables of the child class
* `GridFieldFilterHeader`will work with all variables of the child class

Add the $default_child value via `config.yml`:

```yml
NewsHolder:
  extensions:
    - Lumberjack
  child_classname: 'NewsPage'
```
or via static variable inside the extended Class:

```php
class NewsHolder extends Page {
  private static $extensions = array('Lumberjack');
  private static $child_classname = 'NewsPage'
}
```

Credits to purplespider who developed this addition: https://github.com/micmania1/silverstripe-lumberjack/pull/16

dhensby commented on code/extensions/Lumberjack.php:
```diff
@@ -147,4 +147,20 @@ protected function shouldFilter() {
+	protected function getChildClassName() {
+		$childClassName = "SiteTree";
+		if ($childClassNameConfig = Injector::inst()->create($this->owner->ClassName)->config()->child_classname) {
+			if (class_exists($childClassNameConfig)) {
```
`Config::inst()->get($this->owner->ClassName, 'child_classname` is a bit more concise

tractorcow replied:
Or even `$this->owner->config()->child_classname`?

dhensby commented:
A you're adding a new method, this will need to go against master, I'm afraid

sebastiand commented:
Added a new pull request: https://github.com/silverstripe/silverstripe-lumberjack/pull/34

Considering:
- pull request against master (https://github.com/silverstripe/silverstripe-lumberjack/pull/33#issuecomment-192617431)
- Making getting config more conscise (https://github.com/silverstripe/silverstripe-lumberjack/pull/33#discussion_r55119299)

Also added documentation to README.md
